react-example/src/views/EmbeddedMsgs.tsx

```diff
@@ -142,97 +130,44 @@ export const EmbeddedMsgs: FC<Props> = () => {
         {messages.length > 0 ? (
           messages.map((message: any, index: number) => {
             const data = message?.elements;
-            const primaryButton = data?.buttons?.[0];
-            const secondaryButton = data?.buttons?.[1];
-            const defaultAction = data?.defaultAction;
-
             switch (selectedButtonIndex) {
               case 0:
                 return (
                   <Card
                     key={index.toString()}
-                    cardStyle={{ margin: 0 }}
-                    title={data?.title}
-                    text={data?.body}
-                    imgSrc={data?.mediaUrl}
-                    primaryBtnLabel={primaryButton?.title}
-                    secondaryBtnLabel={secondaryButton?.title}
-                    onClickPrimaryBtn={() =>
-                      handleOpenUrl(
-                        primaryButton?.action?.type,
-                        primaryButton?.action?.data
-                      )
-                    }
-                    onClickSecondaryBtn={() =>
-                      handleOpenUrl(
-                        secondaryButton?.action?.type,
-                        secondaryButton?.action?.data
-                      )
-                    }
-                    onClickView={() =>
-                      handleOpenUrl(defaultAction?.type, defaultAction?.data)
-                    }
+                    parentStyle={{ margin: 0 }}
+                    messageData={data}
+                    handleEmbeddedClick={() => {
+                      return iterableActionRunner.executeAction(
+                        null,
+                        {
+                          type: data?.defaultAction?.type,
+                          data: data?.defaultAction?.data
+                        },
+                        IterableActionSource.EMBEDDED
+                      );
+                    }}
```

Comment:
@hani-iterable I'm not seeing `handleEmbeddedClick` referenced anywhere else? (In the `Card` class for example?) Is this getting used?

Reply:
@hani-iterable confirmed this is being addressed in the other click handling PR